### DISCUSSION TITLE
Dev-help required - Showing user set of options and getting the chosen option

### DISCUSSION THREAD

**Discussion by arsaboo** [beetbox/beets#4790] [Q&A] [answered]
I am trying to improve a plugin and want to show users options to choose from (to pick a track from range of options). 

Specifically, `sorted_tracks` lists candidate tracks that I want the user to pick from. 
 
`sorted_tracks = self.find_closest_match(song['title'], tracks)`

How do I list the tracks in `sorted_tracks` and get the user choice?

I tried `sel = ui.input_options([], numrange=(1, len(sorted_tracks)))` but that does not work. Any guidance will be appreciated

Here's the full context:
https://github.com/arsaboo/beets-plexsync/blob/301990049736c4066d1475f5bd1844bd3e7dca41/beetsplug/plexsync.py#L596

**Comment by sampsyo** [comment]
Hmm, that's odd… can you explain more about what didn't work about your call to `ui.input_options`? That seems like it should do the trick.

**Reply by arsaboo** [reply]
All I get is 
```
# selection (default 1)?
```
There are no tracks listed to choose from.

**Reply by arsaboo** [reply]
So, here's the relevant snippet:
```python
            sorted_tracks = self.find_closest_match(song['title'], tracks)
            self._log.debug('Found {} tracks for {}', len(sorted_tracks),
                            song['title'])
            # present these options to the users and let them choose. How do I do this?
            sel = ui.input_options([], numrange=(1, len(sorted_tracks)))
            print(sel)
```

Even, when there are multiple matches, I only see the following (notice no tracks are listed):
```bash
plexsync: Found 3 tracks for Tum Se Hi
# selection (default 1)?
```
My guess is that I am missing something very basic.

**Reply by govynnus** [reply]
I think where you have an empty list `[]`, you need to give the options to choose from, so maybe
```
sel = ui.input_options(
    [f"{track}".lower() for track in sorted_tracks],
    numrange=(1, len(sorted_tracks)),
)
```
The docstring for [`ui.input_options`](https://github.com/beetbox/beets/blob/9527a07767629c1ceb99c2cd681b78172a7272a0/beets/ui/__init__.py#L207)  might be helpful if you haven't seen it already.

Alternatively [`ui.input_select_objects`](https://github.com/beetbox/beets/blob/9527a07767629c1ceb99c2cd681b78172a7272a0/beets/ui/__init__.py#L377) might be more suitable.

**Reply by arsaboo** [reply, marked as answer]
I guess, we are getting there :)
```python
            print_(f'Choose candidates for { song["album"] } - { song["title"] }:')
            for i, track in enumerate(sorted_tracks, start=1):
                print_(f'{i}. {track.title} - {track.artist().title}')

            sel = ui.input_options(('aBort', 'Enter search', 'Skip'),
                numrange=(1, len(sorted_tracks)), default=1)
```
The output:
```bash
Choose candidates for Piya Basanti - Piya Basanti:
1. Piya Basanti - Sultan Khan
2. Piya Basanti Re - Neha Jad
3. Piya Basanti (Redux) - Achint
4. Piya Basanti (In The Lounge) - Remix by DJ Rishabh - Sultan Khan
# selection (default 1), aBort, Enter search, Skip?
```
